### PR TITLE
Add depth parameter

### PR DESCRIPTION
**Pull request by ClaroHenrique**
This PR aims to add the `depth` parameter in the `GADM.get` function.
This feature is mentioned in [issue 40](https://github.com/JuliaGeo/GADM.jl/issues/40).

### CODE REVIEW

**Comment by juliohm** [comment]
Thank you @ClaroHenrique for starting this PR ❤️ We need to replace the option `children` by the option `level`. When `level=0` that is equivalent to `children=false`, when `level=1` that is equivalent to `children=true` and when `level=2,3,...` that is a new feature with a single table containing all subtables merged into a single one.

Please let me know if something is not clear. I can provide a simple example with states and cities in Brazil.

**Comment by ClaroHenrique** [comment]
Hi @juliohm, I think i get it now.
My main question is about how can we join the subtables in a single one.
In the original database, the data is separated by level in tables with different columns.

<details>
<summary>Schema example</summary>

image

</details>

Is there a problem if the resulting single table hold all those columns?

**Comment by juliohm** [comment]
Hi @ClaroHenrique can you provide an example of result with Brazil data? What are the tables we get when we set level=0,1,2?

We can then try to figure out a combination of columns that makes sense in general.

**Comment by ClaroHenrique** [comment]
@juliohm, those are the results from the database downloaded in [https://data.biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_BRA_gpkg.zip](https://data.biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_BRA_gpkg.zip).

<details> <summary>Level 0</summary> 

image
</details>


<details> <summary>Level 1</summary> 

image
</details>

<details> <summary>Level 2</summary> 

image
</details>

**Comment by juliohm** [comment]
Nice, so the idea is that if a user types

```julia
get("BRA", "Alagoas", level=1)
```

we return all the cities of Alagoas in a single table. This consists of filtering the rows of the table of level 2 that have Alagoas as the state. More generally, we want

```julia
get(country, state, city, municipality, ..., level=n)
```

to return all the leaves of the node country->state->city->municipality that are n steps deeper in the tree.

This could be implemented with different algorithms. We could use a depth-first-search starting from the municipality to find the children at depth n. We then push these tables to a list of tables to be merged. At the end all these tables will be at the same level, so we should expect them to have the same columns. If they don't have the same columns, we should probably investigate the issue further.

I can take a closer look at the dataset if it is still not clear what the end goal is.

**Comment by ClaroHenrique** [comment]
It's fine. I just missunderstood [this](https://github.com/JuliaGeo/GADM.jl/pull/46#issuecomment-1146872422) message.
For a moment I thought that we were going to process multiple levels in one `get`.

File: src/GADM.jl [hunk 1]
```diff
@@ -122,23 +122,19 @@ function get(country, subregions...; children=false)
...
-    plevel = length(subregions)
-    plevel >= nlayers && throw(ArgumentError("more subregions provided than actual"))
-    pname = isempty(subregions) ? "" : last(subregions)
-    player = getlayer(data, plevel)
-    p = filterlayer(player, "NAME_$(plevel)", pname, iszero(plevel))
-    isempty(p) && throw(ArgumentError("could not find required region"))
+    # fetch query params
+    qname = isempty(subregions) ? "" : last(subregions)
+    qlevel = length(subregions)
+    qlevel >= nlayers && throw(ArgumentError("more subregions provided than actual"))
```
Comment (juliohm) [inline src/GADM.jl]:
```suggestion
    qlevel ≥ nlayers && throw(ArgumentError("more subregions provided than actual"))
```

File: test/GADM.jl [hunk 2]
```diff
@@ -36,22 +36,38 @@ end
...
-    # valid country code, children=true
-    parent, children = GADM.get("IND";children=true)
-    @test parent isa NamedTuple
-    @test children isa NamedTuple
-    @test GeoInterface.geomtrait(parent.geom[1]) isa MultiPolygonTrait
-    @test length(children) == 11 #number of fields in named tuple
-    geometries = Tables.getcolumn(children, Symbol("geom"))
+    # valid country code
+    country = GADM.get("IND")
+    @test country isa NamedTuple
```
Comment (juliohm) [inline test/GADM.jl]: I think we want to test that the output satisfies the Tables.jl interface, that is Tables.istable(country). Can you please double check all tests that check against NamedTuple and replace by Tables.istable?

**Review by juliohm — CHANGES_REQUESTED** [review]
@ClaroHenrique this is an amazing PR. Thank you for improving the API, the code is very elegant.

File: test/GADM.jl (hunk 2 above)
Comment (ClaroHenrique) [reply test/GADM.jl]: ok

**Review by juliohm — APPROVED** [review]

**Pull request merged**